src/frontend/views/games_view.py

```diff
@@ -37,6 +38,8 @@ class GamesListByNetworkView(ListView):
   def get_queryset(self):
     self.run = get_object_or_404(Run, name=self.kwargs["run"])
     self.network = get_object_or_404(Network, name=self.kwargs["network"])
+    if self.network.run != self.run:
+      raise Http404("Network exists but is from the wrong run")
```

Comment:
Is 404 the right error for this. I would have said 400, or 409.

Reply:
Same thing - you access a path that has the run, and then a network that doesn't exist for that run - that's a 404 right?